src/hubmap_translator.py

```diff
@@ -734,9 +747,41 @@ def add_datasets_to_entity(self, entity):
 
         logger.info("Finished executing add_datasets_to_entity()")
 
-
-    def entity_keys_rename(self, entity):
-        logger.info("Start executing entity_keys_rename()")
+    # Given a dictionary for an entity containing Neo4j and entity-api-generated data, assume all entries
+    # are to be used in OpenSearch documents.  Modify any key names specified to change, and return a dictionary of
+    # all names and values to store in an OpenSearch document.
+    def _entity_keys_rename(self, entity):
+        global neo4j_to_es_attribute_name_map
+
+        logger.info("Start executing _entity_keys_rename()")
+
+        for remapped_key in neo4j_to_es_attribute_name_map:
+            if remapped_key not in entity:
+                continue
+            attribute_value = entity[remapped_key]
+            # Since statement above assures remapped key is in entity, and the value
+            # is captured, delete the dict entry keyed by the Neo4j name.
+            del entity[remapped_key]
+            # Restore the value to a dict entry keyed by the OpenSearch name.
+            entity[neo4j_to_es_attribute_name_map[remapped_key]] = attribute_value
+
+        # Special case of Sample.rui_location
+        # To be backward compatible for API clients relying on the old version
+        # Also gives the ES consumer flexibility to change the inner structure
+        # Note: when `rui_location` is stored as json object (Python dict) in ES
+        # with the default dynamic mapping, it can cause errors due to
+        # the changing data types of some internal fields
+        # isinstance() check is to avoid json.dumps() on json string again
+        if 'rui_location' in entity and isinstance(entity['rui_location'], dict):
+            entity['rui_location'] = json.dumps(entity['rui_location'])
+
+        logger.info("Finished executing _entity_keys_rename()")
+
+# Given a dictionary for an entity containing Neo4j and entity-api-generated fields, return a
+    # dictionary containing only the fields which will go into an OpenSearch document, with keys renamed to
+    # correspond to OpenSearch document field names.
+    def _entity_keys_rename_and_retain(self, entity):
```

Comment:
Looks like you are only using this `_entity_keys_rename()`, we can delete `_entity_keys_rename_and_retain()` rather than commented out its references if is not being used at all.

Reply:
Done, thank you.  Sorry, too much context switch past 24 hours...